File: frontend/src/api/ratings.ts
import { api } from './client'

export interface RatingResponse {
  id: string
  rater_id: string
  ratee_id: string
  event_id: string
  score: number
  comment: string | null
  created_at: string
  author?: string
  target_name?: string
}

export interface RatingCreatePayload {
  ratee_id: string
  event_id: string
  score: number
  comment?: string | null
}

export interface UserStatsResponse {
  games_played: number
  average_rating: number
  reliability: number
}

export const ratingsApi = {
  getReceived: (skip?: number, limit?: number) =>
    api.get<RatingResponse[]>('/ratings/received', { params: { skip, limit } }),

  getGiven: (skip?: number, limit?: number) =>
    api.get<RatingResponse[]>('/ratings/given', { params: { skip, limit } }),

  create: (payload: RatingCreatePayload) =>
    api.post<RatingResponse>('/ratings', payload),

  getUserStats: (userId: string) =>
    api.get<UserStatsResponse>(`/users/${userId}/stats`),
}

File: backend/app/schemas/ratings.py
"""Pydantic schemas for ratings."""

from datetime import datetime

from pydantic import BaseModel, Field


class RatingCreateRequest(BaseModel):
    ratee_id: str
    event_id: str
    score: float = Field(ge=0.0, le=5.0)
    comment: str | None = Field(default=None, max_length=500)


class RatingResponse(BaseModel):
    id: str
    rater_id: str
    ratee_id: str
    event_id: str
    score: float
    comment: str | None
    created_at: datetime
    author: str | None = None
    target_name: str | None = None

    model_config = {"from_attributes": True}


class UserStatsResponse(BaseModel):
    games_played: int = 0
    average_rating: float = 0.0
    reliability: float = 0.0
